pkg/upgrade/control_plane.go

```diff
@@ -82,8 +82,11 @@ func NewControlPlaneUpgrader(log logr.Logger, config Config) (*ControlPlaneUpgra
 	if config.KubernetesVersion == "" {
 		return nil, errors.New("kubernetes version is required")
 	}
-	if !upgradeIDInputRegex.MatchString(config.UpgradeID) {
-		return nil, errors.New("upgrade ID must be a timestamp containing only digits")
+	if config.UpgradeID == "" {
+		return nil, errors.New("upgrade ID is required")
+	}
+	if errs := validation.IsDNS1123Subdomain(config.UpgradeID); len(errs) > 0 {
```

Comment:
Can you add a comment around why this is dns 1123 subdomain? I can't recall RFC 1123 off the top of my head. I assume this is used because it's conveniently already written and tested?

Reply:
Updated w/a comment